uplink/src/collector/downloader.rs

```diff
@@ -308,46 +300,45 @@ impl FileDownloader {
         Ok((file, file_path))
     }
 
+    // Retry mechanism tries atleast 3 times before returning a download error
```

Comment:
Did we move this function down? It's hard to compare this with the deleted `read_thrice` code above

Reply:
Sagar had implemented retries from when the file was being created, with the changes brought about in this PR, it is obvious that we have to retry only for network errors which only occur during the download and write stage, which is this was move had to be made.

Reply:
Didn't understand. How is this related to moving `retry_thrice` from line `187` to line `304`